backend/app/data_products/router.py

```diff
@@ -189,11 +201,17 @@ def get_signin_url(
     )
 
 
-@router.get("/{id}/conveyor_notebook_url")
+@router.get(
+    "/{id}/conveyor_notebook_url",
+    dependencies=[Depends(OnlyProductRoles([DataProductUserRole.MEMBER]))],
+)
 def get_conveyor_notebook_url(id: UUID, db: Session = Depends(get_db_session)) -> str:
     return DataProductService().get_conveyor_notebook_url(id, db)
 
 
-@router.get("/{id}/conveyor_ide_url")
+@router.get(
+    "/{id}/conveyor_ide_url",
+    dependencies=[Depends(OnlyProductRoles([DataProductUserRole.MEMBER]))],
```

Comment:
It's a bit confusing that you restrict access to a member but allow access to a member and owner. 
I'd suggest providing the role(s) only if you want to restrict access for the specific role, otherwise by default it allow access to member and owner.

Reply:
Great suggestion